core/metacling/src/TClingMethodInfo.h

```diff
@@ -53,35 +53,99 @@ namespace ROOT {
 class TClingClassInfo;
 class TClingTypeInfo;
 
+/// Iteration over Special Functions
+
+class TClingCXXRecMethIter final: public TClingMemberIter {
+
+class SpecFuncIter {
+   using Vec_t = llvm::SmallVector<clang::FunctionDecl *,4>;
+   Vec_t fDefDataSpecFuns; // Special functions materialized from DefinitionData.
+   size_t fIDefDataSpecFuns = 0; // Current element in fDefDataSpecFuns.
+
+public:
+   SpecFuncIter() = default;
+   SpecFuncIter(cling::Interpreter *interp, clang::DeclContext *DC,
+                llvm::SmallVectorImpl<clang::Decl*> &&specFuncs);
+
+   bool Valid() const { return fIDefDataSpecFuns < fDefDataSpecFuns.size(); }
+
+   bool Next() {
+      ++fIDefDataSpecFuns;
+      return Valid();
+   }
+
+   clang::FunctionDecl *operator->() const {
+      return operator*();
+   }
+
+   clang::FunctionDecl *operator*() const {
+      return Valid() ? fDefDataSpecFuns[fIDefDataSpecFuns] : nullptr;
+   }
+};
+
+   SpecFuncIter fSpecFuncIter;
+
+protected:
+   const clang::Decl *
+   InstantiateTemplateWithDefaults(const clang::RedeclarableTemplateDecl *TD) const final;
+
+   bool Skip(const clang::Decl* FD) const final;
+   bool Skip(const clang::UsingShadowDecl* USD) const final;
+
+   clang::Decl *AdvanceUnfiltered() final {
+      if (fSpecFuncIter.Valid() && fSpecFuncIter.Next())
+         return *fSpecFuncIter;
+      return TClingMemberIter::AdvanceUnfiltered();
+   }
+
+public:
+   TClingCXXRecMethIter() = default;
+   TClingCXXRecMethIter(cling::Interpreter *interp, clang::DeclContext *DC,
+                        llvm::SmallVectorImpl<clang::Decl*> &&specFuncs):
+      TClingMemberIter(interp, DC), fSpecFuncIter(interp, DC, std::move(specFuncs)) {
+      Init();
```

Comment:
This may or may not be doing what you want.  If it does, it ought to be explicitly documented.  
Reason: Init uses the virtual function [Is]Valid